### PR TITLE
Add `get_node_id` function.

### PR DESCRIPTION
**Pull request by teymour-aldridge**

### CODE REVIEW

File: src/arena.rs [hunk 1]
```diff
@@ -37,6 +37,10 @@ impl<T> Arena<T> {
     pub fn new() -> Arena<T> {
         Self::default()
     }
+    
+    pub fn get_node_id(node: Node<T>) -> Option<NodeId> where T: PartialEq {
```
Comment (saschagrunert) [inline src/arena.rs:41]: Can you please add a doc comment and example/test for that?

Comment (teymour-aldridge) [reply src/arena.rs:41]: Yes. Sorry, I should have raised an issue and all – was more experimenting to see if this would work.

Comment (saschagrunert) [reply src/arena.rs:41]: Yes take your time, no worries at all. :)

**Comment by teymour-aldridge** [comment]
No problem! I've previously used `indextree` to try and write a [compiler](https://github.com/teymour-aldridge/pseudocompiler) and I raised this PR to implement this feature which I'm intending to use in the [Yew developer tools](https://github.com/yewstack/yew/pull/1221). It's an awesome library; thanks for developing it!

**Comment by saschagrunert** [comment]
Thank you for the feedback, I really appreciate it! I’ll cut a new release tomorrow. :)

**Pull request merged**